test/functional/media_test.rb

```diff
@@ -0,0 +1,130 @@
+require File.join(File.dirname(__FILE__), 'test_helper')
+
+describe 'medium' do
+  describe 'list' do
+    before do
+      @cmd = %w[medium list]
+      @mediums = [{
+        id: 1,
+        name: 'CentOS mirror',
+        path: 'http://mirror.centos.org/centos/$major/os/$arch',
+        os_family: 'Redhat',
+        operating_systems: [],
+        locations: [],
+        organizations: []
+      }]
+    end
+
+    it 'should return a list of mediums' do
+      api_expects(:media, :index, 'List mediums').returns(@mediums)
```

Comment:
nit: do you think it would be reasonable to use `media` as a plural so that we are consistent? 

Also one line above in test name.